From 45c6c7ba1c4673750df2c6f6c041764f312f6b57 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Mon, 14 Dec 2020 20:28:18 +0200
Subject: [PATCH] [SW-0] gaudi: security: retrieve check on security

The check on 'security enabled' in function
preboot_security_cleanup_before_run () was mistakenly
deleted in a prior commit.

Change-Id: I634adba6aa4d9b081cde15fc242e85b6f479add2
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index c49e8d4..89d6389 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1495,6 +1495,8 @@
 
 void preboot_security_cleanup_before_run(void)
 {
+	if (!preboot_security_is_enabled())
+		return;
 
 	/* Enable the SIF7 HBW SRAM_IF_DYN_RR_BGN - SRAM_IF_DYN_RR_END */
 	preboot_secure_hbw_sram();
